boxoffice/models/discount_policy.py

```diff
@@ -0,0 +1,75 @@
+from boxoffice.models import db, BaseScopedNameMixin, IdMixin
+from datetime import datetime
+from boxoffice.models import ItemCollection
+from baseframe import __
+from coaster.utils import LabeledEnum
+
+import string
+import random
+
+__all__ = ['DiscountPolicy', 'DiscountCoupon', 'item_discount_policy']
+
+
+def generate_coupon_code(size=6, chars=string.ascii_uppercase + string.digits):
+    return ''.join(random.choice(chars) for _ in range(size))
```

Comment:
Item declaration order in a module should be imports, constants, variables, and then functions and classes. This should go below `item_discount_policy`.